### PR TITLE
Implement error Status

### PR DESCRIPTION
**Pull request by Thumimku**
## Purpose
- Implement unique resource URI
- Implement Error status functionality.

## Approach
- Resource URI to avoid duplicate remote fetch configuration for one remote repo.
- Error status is updated while resolving application name and deploying.

## Tested with
- H2
- MySQL
- MSSQL
- DB2
- POSTGRESQL

### CODE REVIEW

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java [hunk 1]
```diff
@@ -238,15 +257,45 @@ public void setActionListenerAttributes(Map<String, String> actionListenerAttrib
...
+     */
+    public String getRemoteResourceURI() {
+
+        return remoteResourceURI;
+    }
+
+    /**
+     * derive RemoteResourceURI from RepositoryManagerAttributes.
+     */
+    public void deriveRemoteResourceURI() {
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java:282]: Better we calculate this with a setter.

Comment (ruwanta) [reply components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java:282]: calling derive RemoteResourceURI() from the client code does not look elegant

File: features/org.wso2.carbon.identity.remotefetch.core.feature/resources/dbscripts/postgresql.sql [hunk 2]
```diff
@@ -20,6 +22,7 @@ CREATE TABLE IDN_REMOTE_FETCH_REVISIONS (
 	DEPLOYED_DATE TIMESTAMP,
 	DEPLOYMENT_STATUS VARCHAR(255),
 	ITEM_NAME VARCHAR(255),
+	DEPLOYMENT_ERROR TEXT,
```
Comment (ruwanta) [inline features/org.wso2.carbon.identity.remotefetch.core.feature/resources/dbscripts/postgresql.sql:25]: can we change this field to, 
DEPLOY_ERR_LOG

Reasons,
1. It has a part LOG which says it is a log
2. Column name made short

File: components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/java/org/wso2/carbon/identity/remotefetch/core/ui/client/RemoteFetchConfigurationClient.java [hunk 3]
```diff
@@ -153,6 +153,7 @@ private static RemoteFetchConfiguration parseJsonToConfiguration(String jsonObje
 
         Gson gson = new Gson();
         RemoteFetchConfiguration fetchConfiguration = gson.fromJson(jsonObject, RemoteFetchConfiguration.class);
+        fetchConfiguration.setRemoteResourceURI(fetchConfiguration.getRepositoryManagerType());
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.remotefetch.core.ui/src/main/java/org/wso2/carbon/identity/remotefetch/core/ui/client/RemoteFetchConfigurationClient.java:156]: It looks wrong to set Remote Resource URI as a Repository manager type.
Looks still the method name does not correspond to what it supposed to do

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java [hunk 4]
```diff
@@ -238,15 +257,45 @@ public void setActionListenerAttributes(Map<String, String> actionListenerAttrib
...
+        return remoteResourceURI;
+    }
+
+    /**
+     * Set RemoteResourceURI from RepositoryManagerAttributes.
+     */
+    public void setRemoteResourceURI(String repositoryManagerType) {
+
+        if (this.repositoryManagerType.equals(repositoryManagerType)) {
+            String repoURI = this.repositoryManagerAttributes.get("uri");
```
Comment (ruwanta) [inline components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/RemoteFetchConfiguration.java:285]: What happens when repositoryManagerAttributes == null?

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/BasicRemoteFetchConfiguration.java [hunk 5]
```diff
@@ -36,6 +36,23 @@
     private int failedDeployments;
     private Date lastDeployed = null;
 
+    /**
+     * Default Constructor used by DAO.
+     */
+    public BasicRemoteFetchConfiguration() {
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/BasicRemoteFetchConfiguration.java:42]: add new line after method signature

File: components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/BasicRemoteFetchConfiguration.java [hunk 6]
```diff
@@ -36,6 +36,23 @@
...
+    }
+
+    /**
+     * Following constructor is used only by Test classes.
+     * @param id
+     * @param isEnabled
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/BasicRemoteFetchConfiguration.java:48]: add paramter descriptions

File: components/org.wso2.carbon.identity.remotefetch.core/src/test/resources/dbScripts/permission.sql [hunk 7]
```diff
@@ -18,7 +20,8 @@ CREATE TABLE IF NOT EXISTS IDN_REMOTE_FETCH_REVISIONS (
 	DEPLOYED_DATE TIMESTAMP,
 	DEPLOYMENT_STATUS VARCHAR(255),
 	ITEM_NAME VARCHAR(255),
+	DEPLOY_ERR_LOG VARCHAR(MAX),
 	PRIMARY KEY (ID),
 	FOREIGN KEY (CONFIG_ID) REFERENCES IDN_REMOTE_FETCH_CONFIG(ID) ON DELETE CASCADE,
 	CONSTRAINT UC_REVISIONS UNIQUE (CONFIG_ID,ITEM_NAME)
-);
+);
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.remotefetch.core/src/test/resources/dbScripts/permission.sql:27]: Add new line at end of the file

**Pull request merged**